### PR TITLE
tracing: Add support for B3 headers via the JAEGER_PROPAGATION env var

### PR DESCRIPTION
**Pull request by ptescher**
This will provide compatibility with istio.

Fixes https://github.com/ory/hydra/issues/1447

### CODE REVIEW

File: tracing/tracer.go [hunk 1]
```diff
@@ -52,8 +54,21 @@ func (t *Tracer) Setup() error {
 			jc.Reporter.LocalAgentHostPort = t.JaegerConfig.LocalAgentHostPort
 		}
 
+		var configs []jeagerConf.Option
+
+		// This works in other jaeger clients, but is not part of jaeger-client-go
+		if os.Getenv("JAEGER_PROPAGATION") == "b3" {
```
Comment (aeneasr) [inline tracing/tracer.go]: Please, instead:

1. Add `Propagation string` to [`JaegerConfig`](https://github.com/ory/hydra/pull/1456/files#diff-6aa771dfea3dd559e0b2c9263ebf1ccdR25)
2. Use the `Propagation` value here (`t.JaegerConfig`)
3. Set the value [here](https://github.com/ory/hydra/blob/master/driver/configuration/provider_viper.go#L243)
4. Document the new value [around here](https://github.com/ory/hydra/blob/master/docs/config.yaml#L399)
To stay compatible it probably makes sense to do something like (assuming `JAEGER_PROPAGATION` is a de-facto standard):

```
Propagation: stringsx.Coalesce(
  viper.GetString("JAEGER_PROPAGATION")
  viperx.GetString(v.l, "tracing.providers.jaeger.propagation", ""),
),
```

Comment (ptescher) [reply tracing/tracer.go]: Pushed some new changes, let me know if that works. I also added `TRACING_PROVIDER_JAEGER_PROPAGATION` just for consistency.

**Comment by aeneasr** [comment]
Your changes to ory/x are now released as v0.0.61

**Comment by aeneasr** [comment]
ps: If you rebase on master, the module changes should be up to 0.0.61

**Comment by ptescher** [comment]
@aeneasr all done, tests are passing now.

**Comment by aeneasr** [comment]
Awesome, thank you!

**Pull request merged**